application/cms/models.py

```diff
@@ -43,6 +42,12 @@
     db.Column("page_id", db.String, primary_key=True),
 )
 
+subtopic_measure = db.Table(
```

Comment:
Yeah I see this format is used in a couple of places, but I think I prefer the form used in most other places, which is the `class SubtopicMeasure(db.Table)` form. Could we do that instead - or is there a good reason for this form?

Reply:
Also I think this could benefit from being more proximal to the subtopic/measure table.

Reply:
I guess my reasoning behind this way of doing things is that we don't really ever want to adress this table directly as a class in its own right - it isn't part of our "data model" as such, but is more a kind of "helper" that allows the many-to-many relationship between Subtopic and Measure to be defined.  Making it a class implies that we consider a "SubtopicMeasure" to be a thing we might want to instantiate a an object, and that doesn't seem quite right to me.

In terms of proximity, yep if it's going to be a class then definitely it would live next to the subtopic and measure class definitions, doing it this way it needs to come before the relationship definitions that use it, and so near the top of the file seemed better (and is where other "constant" tables are defined).

Reply:
Hmm, yeah we have a mix - of the association tables DataSourceInPage is the only one that has a class of its own.
The other ones are `association_table` and `parent_association_table` that enable the classification relationships.

I think I like this way of doing things - these lookup tables are things that should never be referenced from outside of the model classes, and so not having them as full-blown model objects seems nice.

I could change DataSourceInPage to work the same way for consistency?

Or are you unconvinced by the argument for doing it this way?

Reply:
Actually please leave `DataSourceInPage` - I'll do it. Because I'm changing that in my own stuff.

Reply:
They can't go in between the joined classes, as need to be defined before they are referenced. But can go just above I reckon.

Reply:
You can reference the table by string representations, which I think we moved towards doing? (at least, we recently changed columns from instances to string values)